WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java

```diff
@@ -1,87 +1,131 @@
-    void setIsOldVersion(Boolean outdated);
+    public abstract void setIsOldVersion(Boolean outdated);
 
 	/** @return Whether this client version is outdated. */
-    Boolean isBigUpdateAvailable();
+    public abstract Boolean isBigUpdateAvailable();
 
 	/** Sets whether this client version has a big update available. */
-    void setBigUpdateAvailable(Boolean available);
+    public abstract void setBigUpdateAvailable(Boolean available);
 
 	/** @return Whether the user answered to the big update question. */
-    Boolean isBigUpdateAnswered();
+    public abstract Boolean isBigUpdateAnswered();
 
 	/** Sets whether this client version has a big update available. */
-    void setBigUpdateAnswered(Boolean answered);
+    public abstract void setBigUpdateAnswered(Boolean answered);
 
-    long getIntervals();
+    public abstract long getIntervals();
 
 	/** Adds the number to the transfered intervals for the store. */
-    void addTransferedIntervals(long number);
+    public abstract void addTransferedIntervals(long number);
 
 	/** @return The number of successfully transfered intervals. */
-    String getLastIntervalTransferDate();
+    public abstract String getLastIntervalTransferDate();
 
 	/** Adds the number to the transfered intervals for the store. */
-    void setLastTransferedInterval();
+    public abstract void setLastTransferedInterval();
 
 	/** @return The serverURL. */
-    String getServerURI();
+    public abstract String getServerURI();
 
 	/**
 	 * @return <code>true</code> if this workspace has already been registered
 	 *         with WatchDog, <code>false</code> otherwise. Note: This does not
 	 *         say whether WatchDog should be activated.}.
 	 */
-    boolean isProjectRegistered(String project);
+    public boolean isProjectRegistered(String project) {
+    	ProjectPreferenceSetting projectSetting = getProjectSetting(
+				project);
+		return (projectSetting != null
+				&& projectSetting.startupQuestionAsked);
+    }
 
 	/**
 	 * @return The matching {@link ProjectPreferenceSetting}, or a completely
 	 *         new one in case there was no match.
 	 */
-    ProjectPreferenceSetting getOrCreateProjectSetting(String project);
+    public ProjectPreferenceSetting getOrCreateProjectSetting(String project) {
+    	ProjectPreferenceSetting setting = getProjectSetting(project);
+		if (setting == null) {
+			setting = new ProjectPreferenceSetting();
+			setting.project = project;
+			projectSettings.add(setting);
+		}
+		return setting;
+    }
+    
+    /**
+	 * @return The matching {@link ProjectPreferenceSetting}, or
+	 *         <code>null</code> in case there was no match.
+	 */
+	private ProjectPreferenceSetting getProjectSetting(String project) {
+		for (ProjectPreferenceSetting setting : projectSettings) {
+			if (setting.project.equals(project)) {
+				return setting;
+			}
+		}
+		return null;
+	}
 
 	/**
 	 * Registers the given workspace with WatchDog. If use is <code>true</code>,
```

Comment:
Replace workspace with project? Could you grep the whole code for occurrences of workspace? That should only appear in Eclipse plugin. Ideally, a short explanation on this difference should be in ...DogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/eclipse/ui/preferences/Preferences, for example in the JDoc.